### PR TITLE
Added expandByObject to THREE.Box3

### PR DESCRIPTION
**Pull request by Wilt**
It was not possible to expand a `THREE.Box3` instance for adding objects to an existing bounding-box. By simply moving the `this.makeEmpty()` call outside of `setFromObject` it was simple to add such functionality with a minimum of code.

The method name `expandByObject` corresponds to methods like `expandByPoint`, `expandByVector`, and `expandByScalar`.

### CODE REVIEW

File: src/math/Box3.js [hunk 1]
```diff
@@ -85,11 +85,21 @@ THREE.Box3.prototype = {
...
+
+		return this;
+
+	},
+
+	expandByObject: function () {
+
 		var v1 = new THREE.Vector3();
 
 		return function setFromObject( object ) {
```
Comment (WestLangley) [inline src/math/Box3.js]: This should now be

    return function expandByObject( object ) {

**Comment by WestLangley** [comment]
Do you mind updating the docs, too?

**Comment by WestLangley** [comment]
I should also issue the reminder that we still have the [long-standing problem](https://github.com/mrdoob/three.js/pull/6448#issuecomment-95624064) with `Object3D.updateMatrixWorld()`, which does not do what the name implies because it traverses down the tree rather than up it.

**Comment by Wilt** [comment]
*Docs and function name updated*

**Comment by bhouston** [comment]
I prefer Box3 not to know about Object3D.  Instead I'd just make it easy to get a Box3 from an Object3D and then we can use the existing Box3 classes for initializing from a Box3 and unioning an existing Box3.  I do not generally like math libraries becoming away of high level classes because it is mostly just convenience functions and it is hard to know when to stop -- it leads in my opinion to unnecessary growth in the library for just convenience functions, not truly new functionality.  In my ultimate design of Three.Math there would be no awareness at all of Object3D or Geometry or anything outside of Three.Math.

**Comment by Wilt** [comment]
@bhouston, I understand your comment, but this method is only a spin off of an already existing 'setFromObject' method, so there is actually nothing new introduced in this PR.

**Comment by bhouston** [comment]
@Wilt I understand.  I am just sharing my opinion that I would remove all leaks of Object3D into the math library -- I think that would be more modular and encapsulating in terms of design.  I like modular designs were basic modules (like the math library) are not aware of the higher level constructs (like Object3D).  Generally following these types of guidelines reduces unnecessary interconnections within a program and reduces its complexity.

People that know me know that I hate coupling, and including Object3D into the math libraries is a form of coupling that I view as unnecessary and problematic:

https://en.wikipedia.org/wiki/Coupling_(computer_programming)
https://en.wikipedia.org/wiki/Cohesion_(computer_science)

Having Object3D used directly in the math library means that tree shkaing as implemented by rollup will include massive amounts of code outside of the math library if you happen to use Box3 now.  This is also undesirable.

I won't prevent this PR from being merged, I just I think it is wrong to have Object3D based used by the math library for a bunch of reasons.

I will recuse myself from this PR now that I said the above.  I do not want to be a blocker.

**Comment by WestLangley** [comment]
@bhouston How about these methods, instead?
```javascript
Object3D.setBox( box ); // replaces Box3.setFromObject( object )

Object3D.expandBox( box ); // instead of the proposed Box3.expandByObject( object )
```

**Comment by Wilt** [comment]
I like it. Is the box an optional parameter? Then it would maybe be good to name it `getBox`. And should `setBox` and and `expandBox` take a second parameter for recursive or will it always be recrecursive. 
Maybe it would be nice to add `expandFromPoints` on top of `setFromPoints` to solve the implementation on the `THREE.Box3` side. And those methods could be renamed to `setFromVertices` and `expandFromVertices` to get more in line with the three.js Math class naming. Then the names are even more intuitive.

**Comment by Wilt** [comment]
I could update my PR for this, but I won't manage today...

**Comment by WestLangley** [comment]
>Is the box an optional parameter?

Absolutely not.

>I could update my PR for this

Please wait for more feedback before you go down that path.

**Comment by Wilt** [comment]
When I tried to add tests to this `expandByObject` I already noticed the issues mentioned by @bhouston. For getting those test running. I needed to include the additional dependencies (Mesh, Object3D, Geometry, BufferGeometry) for making the test run correctly.

**Comment by Mugen87** [comment]
see #10474

**Pull request closed without merging**